src/libraries/System.Private.CoreLib/src/System/Threading/Mutex.Windows.cs

```diff
@@ -82,10 +109,41 @@ private static OpenExistingResult OpenExistingWorker(string name, out Mutex? res
         // in a Mutex's ACL is MUTEX_ALL_ACCESS (0x1F0001).
         public void ReleaseMutex()
         {
-            if (!Interop.Kernel32.ReleaseMutex(SafeWaitHandle))
+            bool success = true;
+            string? systemCallErrors;
+            Marshal.BeginTrackingSystemCallErrors();
+            try
             {
-                throw new ApplicationException(SR.Arg_SynchronizationLockException);
+                success = Interop.Kernel32.ReleaseMutex(SafeWaitHandle);
             }
+            finally
+            {
+                systemCallErrors = Marshal.EndTrackingSystemCallErrors(getSystemCallErrors: !success);
+            }
+
+            if (!success)
+            {
+                SystemException? innerEx =
+                    string.IsNullOrEmpty(systemCallErrors) ? null : GetInnerExceptionForSystemCallErrors(systemCallErrors);
+                throw new ApplicationException(SR.Arg_SynchronizationLockException, innerEx);
```

Comment:
Nothing else in the repo uses SystemException for this purpose. The error details are included in the error message in situations like this. Prior art:

https://github.com/dotnet/runtime/blob/b945aeda8a827a54eaf4e91183cf3725c1e090fa/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.NativeAot.Unix.cs#L43-L45

https://github.com/dotnet/runtime/blob/b945aeda8a827a54eaf4e91183cf3725c1e090fa/src/libraries/Common/src/Interop/Unix/Interop.IOErrors.cs#L181-L183